lib/wallet/wallet.js

```diff
@@ -1781,6 +1786,9 @@ class Wallet extends EventEmitter {
       if (!coin)
         continue;
 
+      if (acct && !await this.txdb.hasCoinByAccount(acct, hash, index))
```

Comment:
I might be slipping - but this is an acct index at this point right? Does that mean it could be 0 - and falsey - for the default account?